### PR TITLE
Restore-NuGet-Packages retry handling

### LINKED ISSUE #881: Add cake-build build.cake NuGet restore retry handling
We currently experience allot of hangs/timeouts/etc. during NuGet restore on Linux / OSX build agents, forcing us often to retry PR builds several times.

Cake v0.11.0 introduced a timeout setting to all tools using this and an something like Polly we should be able to do retries while restoring packages and eliminate some frustration.

I'll shortly submit a WIP PR with to implement this.

### PR DESCRIPTION
**Pull request by devlead**
* Added polly addin
* Added Restore policy & handling when invoking NuGetRestore alias
* Added Tooltimeout to the NuGetRestore NuGetRestoreSettings

This fixes  #881

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Install addins.
+#addin "nuget:?package=Polly&version=4.2.0"
+using Polly;
 // Install tools.
```
Comment (patriksvensson) [inline build.cake]: Can we add a space before this?

Comment (devlead) [inline build.cake]: Adjusted.

**Comment by devlead** [comment]
It worked! Cool :sunglasses: , lowered timeout & addressed feedback.

**Comment by patriksvensson** [comment]
@devlead Ok, will merge after CI pass.

**Comment by devlead** [comment]
@patriksvensson pushed again, interestingly `Setup(Action<ICakeContext>)` doesn't seem to work on mono. Will post an issue.

**Comment by devlead** [comment]
@cake-build/team CI passes :wink:  and if you see the [Mac Travis build](https://travis-ci.org/cake-build/cake/jobs/129953222) you can see that this "saved" the build from failing.
It took 2 retries to sort things out which can be seen in build log [line 206](https://travis-ci.org/cake-build/cake/jobs/129953222#L206) and [line 275](https://travis-ci.org/cake-build/cake/jobs/129953222#L275) :+1:

**Comment by patriksvensson** [comment]
Merged! :+1:

**Pull request merged**